find-trio-denovo.py

```diff
@@ -14,167 +13,105 @@
+                                    if is_variant:
+                                        variant_count += 1
+                                        variantFile.write(line)
+
+                                    if is_unphased:
+                                        unphased_count += 1
+                                        unphasedFile.write(line)
+
+                    completion_msg = "\nThe script took {0} minutes\n".format((time.time() - startTime/60))
+
+                    variantFile.write(completion_msg)
+                    variantFile.write("{0} phased and unphased variants were found".format(variant_count))
+
+                    unphasedFile.write(completion_msg)
+                    unphasedFile.write("{0} variants were found when all lines were treated as unphased".format(unphased_count))
 
 ################################ CREATE FILE TO STORE DENOVO VARIANT INFO ####################################
 #############################################################################################################
 
-def recordVariant(line): 
-	file = open(sys.argv[1]+".variants" , "a") #open output file in "append" mode
-	file.write(line) 						   #includes both phased and unphased variants
-	file.close()
+def process_line(line, dadIdx, momIdx, childIdx, variant_count, unphased_count):
+        is_unphased = False
+        is_variant = False
+
+        (chrom, pos, ID, ref, alt, qual, Filter, info, format, samples) = line.strip("\n").split("\t", 9)
+        samples = samples.split("\t")
+        dadgeno = samples[dadIdx]
+        momgeno = samples[momIdx]
+        childgeno = samples[childIdx]
+
+        (dadAlleles, dadPhased) = extract_genes(dadgeno)
+        (momAlleles, momPhased) = extract_genes(momgeno)
+        (childAlleles, childPhased) = extract_genes(childgeno)
+
+        # assumes that delimiter on dadGeno is same as momGeno and childGeno
+        if dadPhased != momPhased or dadPhased != childPhased:
+            print("Family phasing doesn't match!")
+            exit(1)
+
+        phased = dadPhased
+
+        child_in_dad = child_in_parent(childAlleles, dadAlleles)
+        child_in_mom = child_in_parent(childAlleles, momAlleles)
+
+        #The following conditions were written when I was taking all 4 nucleotides into account
+        #In a VCF, it will usually only be comparing a "0" (ref) or "1" (alt)
+        #It still works, but might be more than what's necessary
+
+        #Treat all lines as unphased and record variants in second output file
+        if ((not child_in_dad[0] and not child_in_mom[0]) or # variant because child1 not in either parent
+            (not child_in_dad[1] and not child_in_mom[1]) or # variant because child2 not in either parent
+            (not child_in_dad[0] and not child_in_dad[1]) or # variant because neither child allele in dad
+            (not child_in_mom[0] and not child_in_mom[1])): # variant because neither child allele in mom
```

Comment:
I'm not sure this conditional can be simplified any further. I drew a Karnaugh map to try and reduce it and I'm pretty sure this is as good as it gets.

![untitled spreadsheet - google sheets 2016-03-28 00-31-32](https://cloud.githubusercontent.com/assets/3655/14071627/83ba9cae-f47c-11e5-8848-b19df3e06d03.png)